crates/service/src/service.rs

```diff
@@ -68,64 +55,119 @@ pub async fn run() -> anyhow::Result<()> {
-            },
-            EmptySubscription,
+    let network_subgraph = create_subgraph_client(
+        http_client.clone(),
+        &config.graph_node,
+        &config.subgraphs.network.config,
+    )
+    .await;
+
+    let escrow_subgraph = create_subgraph_client(
+        http_client.clone(),
+        &config.graph_node,
+        &config.subgraphs.escrow.config,
+    )
+    .await;
+
+    // Establish Database connection necessary for serving indexer management
+    // requests with defined schema
+    // Note: Typically, you'd call `sqlx::migrate!();` here to sync the models
+    // which defaults to files in  "./migrations" to sync the database;
+    // however, this can cause conflicts with the migrations run by indexer
+    // agent. Hence we leave syncing and migrating entirely to the agent and
+    // assume the models are up to date in the service.
+    let database =
+        database::connect(config.database.clone().get_formated_postgres_url().as_ref()).await;
+
+    let domain_separator = tap_eip712_domain(
+        config.blockchain.chain_id as u64,
+        config.blockchain.receipts_verifier_address,
+    );
+
+    let router = ServiceRouter::builder()
+        .database(database)
+        .domain_separator(domain_separator)
+        .graphnode_state(graphnode_state)
+        .release(release)
+        .config(config)
+        .network_subgraph(network_subgraph)
+        .escrow_subgraph(escrow_subgraph)
+        .build();
+
+    serve_metrics(config.metrics.get_socket_addr());
+
+    info!(
+        address = %config.service.host_and_port,
+        "Serving requests",
+    );
+    let listener = TcpListener::bind(&config.service.host_and_port)
+        .await
+        .expect("Failed to bind to indexer-service port");
+
+    let app = router.create_router().await?;
+    let router = NormalizePath::trim_trailing_slash(app);
+    //
+    let service = ServiceExt::<Request>::into_make_service(router);
+    Ok(serve(listener, service)
+        .with_graceful_shutdown(shutdown_signal())
+        .await?)
+}
+
+async fn create_subgraph_client(
+    http_client: reqwest::Client,
+    graph_node: &GraphNodeConfig,
+    subgraph_config: &SubgraphConfig,
+) -> &'static SubgraphClient {
+    Box::leak(Box::new(
+        SubgraphClient::new(
+            http_client,
+            subgraph_config.deployment_id.map(|deployment| {
+                DeploymentDetails::for_graph_node_url(
+                    graph_node.status_url.clone(),
+                    graph_node.query_url.clone(),
+                    deployment,
+                )
+            }),
+            DeploymentDetails::for_query_url_with_token(
+                subgraph_config.query_url.clone(),
+                subgraph_config.query_auth_token.clone(),
+            ),
         )
-        .data(agreement_store)
-        .data(prices)
-        .finish();
+        .await,
+    ))
+}
+
+pub async fn shutdown_signal() {
```

Comment:
Please document this or rename it. Without reading the code, I assumed this would be sending a shutdown signal, not setting up for one and awaiting it.